Packs/DarktraceASM/Integrations/DarktraceASMRisk/DarktraceASMRisk.py

```diff
@@ -0,0 +1,1010 @@
+    :param asset: JSON asset as returned by API.
+    :return: Reformatted JSON asset.
+    :rtype: ``Dict[str, Any]``
+    """
+    new_json: Dict[str, Any] = {}
+    for key in asset:
+        if key == 'comments':
+            if asset[key] is None:
+                new_json[key] = {}
+            else:
+                comments = {comment['id']: comment['text'] for comment in asset[key]}
+                new_json[key] = comments
+        elif key == 'discoverySources':
+            if asset[key] is None:
+                new_json[key] = {}
+            else:
+                sources = {source["id"]: source["description"] for source in asset[key]}
+                new_json[key] = sources
+        elif key == 'risks':
+            if asset[key] is None:
+                new_json[key] = {}
+            else:
+                risks = {risk["id"]: risk["title"] for risk in asset[key]}
+                new_json[key] = risks
+        elif key in ['fqdns', 'technologies', 'registeredDomain']:
+            if asset[key] is None:
+                new_json[key] = {}
+            else:
+                values = {value["id"]: value["name"] for value in asset[key]}
+                new_json[key] = values
+        elif key in ['ipaddresses', 'ipAddresses', 'resolvesTo']:
+            if asset[key] is None:
+                new_json[key] = {}
+            else:
+                addresses = {ip["id"]: ip["address"] for ip in asset[key]}
+                new_json[key] = addresses
+        else:
+            new_json[key] = asset[key]
+    return new_json
+
+
+def _compute_xsoar_severity(security_rating: str) -> int:
+    """Translates Darktrace ASM security rating into XSOAR Severity.
+    :type security_rating: ``str``
+    :param security_rating: ASM security rating to convert.
+    :return: Integer equivalent of XSOAR severity scores.
+    :rtype: ``int``
+    """
+    if security_rating in ['c', 'd']:
+        return 2
+    if security_rating in ['e']:
+        return 3
+    if security_rating in ['f']:
+        return 4
+    return 1
+
+
+"""*****COMMAND FUNCTIONS****"""
+
+
+def test_module(client: Client, first_fetch_time: float) -> str:
+    """
+    Returning 'ok' indicates that the integration works like it is supposed to. Connection to the service is successful.
+
+    :type client: ``Client``
+    :param client:
+        Darktrace Client.
+    :type first_fetch_time: ``float``
+    :param first_fetch_time:
+        First fetch time.
+    :return:
+        A message to indicate the integration works as it is supposed to.
+    :rtype: ``str``
+    """
+    try:
+        first_fetch_datetime = datetime.fromtimestamp(first_fetch_time)
+        client.get_asm_risks(start_time=first_fetch_datetime)
+
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+    return 'ok'
+
+
+def fetch_incidents(client: Client,
```

Comment:
In order to take into consideration milliseconds, then we must use a different time format. The original **DATE_FORMAT** will drop the milliseconds.
We can use a date format specific for fetching incidents, for example **FETCH_DATE_FORMAT = "%Y-%m-%dT%H:%M:%S:%f"**, which will take into consideration milliseconds, and use this format in the **fetch_incidents** function.

Reply:
We just added the changes for this. Thanks!